plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -314,12 +314,15 @@ void state_history_plugin::set_program_options(options_description& cli, options
 
 void state_history_plugin_impl::plugin_initialize(const variables_map& options) {
    try {
-      EOS_ASSERT(options.at("disable-replay-opts").as<bool>(), plugin_exception,
-                 "state_history_plugin requires --disable-replay-opts");
-
       chain_plug = app().find_plugin<chain_plugin>();
       EOS_ASSERT(chain_plug, chain::missing_chain_plugin_exception, "");
       auto& chain = chain_plug->chain();
+
+      if (!options.at("disable-replay-opts").as<bool>() && options.at("chain-state-history").as<bool>()) {
```

Comment:
I would have thought it should be: `if (options.at("disable-replay-opts")...` without the `!`.

Reply:
No need to set it if it is already set.

Reply:
I thought this was an option passed to ship and I didn't realize it would already be applied to `chain`. Thanks.